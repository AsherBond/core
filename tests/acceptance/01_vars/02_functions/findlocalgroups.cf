body common control
{
  inputs => { "../../default.cf.sub" };
  bundlesequence => { default("$(this.promise_filename)") };
  version => "1.0";
}

bundle agent init
{
  vars:
  # simple filters
    "simple_filter"
      slist => { "name=root" };
    "number_filter"
      slist => { "gid=0" };
  # longer filters
    "slist_filter"
      slist => { "gid=0", "name=root" };
  # using data
    "data_filter"
      data => '[ "gid=0", "name=root" ]';
  # using regex
    "simple_regex"
      slist => { "name=roo.*" };
    "number_regex"
      slist => { "gid=0.*" };
    "longer_regex"
      slist => { "name=ro.*", "gid=0.*" };
  # non-existent group
    "unknown"
      slist => { "name=thisgroupdoesntexist" };
}

bundle agent test
{
  meta:
    "test_soft_fail"
      string => "!linux|(termux|android)",
      meta => { "CFE-2318" };
  vars:
    "glist1"
      data => findlocalgroups("@(init.simple_filter)");
    "glist2"
      data => findlocalgroups("@(init.number_filter)");
    "glist4"
      data => findlocalgroups("@(init.slist_filter)");
    "glist3"
      data => findlocalgroups("@(init.data_filter)");
    "glist5"
      data => findlocalgroups("@(init.simple_regex)");
    "glist6"
      data => findlocalgroups("@(init.number_regex)");
    "glist7"
      data => findlocalgroups("@(init.longer_regex)");
    "glist8"
      data => findlocalgroups("@(init.unknown)");
}

bundle agent check
{
  methods:
    "check"
      usebundle => dcs_check_state(
        test,
        "$(this.promise_filename).expected.json",
        $(this.promise_filename)
      );
}
