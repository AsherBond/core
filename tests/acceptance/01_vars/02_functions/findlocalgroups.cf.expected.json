{
  "glist1": {
    "root": {
      "gid": 0,
      "members": []
    }
  },
  "glist2": {
    "root": {
      "gid": 0,
      "members": []
    }
  },
  "glist3": {
    "root": {
      "gid": 0,
      "members": []
    }
  },
  "glist4": {
    "root": {
      "gid": 0,
      "members": []
    }
  },
  "glist5": {
    "root": {
      "gid": 0,
      "members": []
    }
  },
  "glist6": {
    "root": {
      "gid": 0,
      "members": []
    }
  },
  "glist7": {
    "root": {
      "gid": 0,
      "members": []
    }
  },
  "glist8": {
  }
}
